lib/squadster/formations/services/create_squad_request.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Squadster.Formations.Services.CreateSquadRequest do
+  alias Squadster.Repo
+  alias Squadster.Formations.SquadRequest
+
+  def call(squad_id, user) do
+    %{squad_request: existing_request, squad_member: squad_member} =
+      user
+      |> Repo.preload(:squad_request)
+      |> Repo.preload(:squad_member)
+
+    if is_nil(squad_member) do
+      create(user, squad_id, existing_request)
+    else
+      {:error, "User already has squad"}
+    end
+  end
+
+  defp create(user, squad_id, existing_request) do
```

Comment:
I think it'll be better to use structs and pattern match their ids because in functions level higher we operating structs, and as I said below passing structs looks cleaner than `struct.id`

Reply:
Nice, seems pretty good